pkg/engine/journal_snapshot.go

```diff
@@ -268,6 +279,10 @@ func (sm *JournalSnapshotManager) Write(base *deploy.Snapshot) error {
 	return sm.journal.AddJournalEntry(je)
 }
 
+func (sm *JournalSnapshotManager) RebuiltBaseState() error {
```

Comment:
would love to have some validation/assertions here, too, to help catch bugs in the generator

Reply:
Hrm we currently don't keep track of the journal entries in the snapshot manager at all.  To add meaningful validations we'd have to keep track of all journal entries. It's also impossible to tell if this is the last journal entry at this point, so not quite sure what validations we should add here.

Reply:
Made the suggested fixes; will push to this branch